### ISSUE TITLE
Ceph: mon scheduling bug!

### ISSUE THREAD

**Issue by robinbraemer** [rook/rook#3501]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Mon pod schedule on k8s nodes that are not fulfilling specified (in CephCluster config) requirements (such as enough RAM)

**Expected behavior:**
[scheduleMonitor func](https://github.com/rook/rook/blob/50ef726ad34da4f2683d5f4304eb4e7910bdb938/pkg/operator/ceph/cluster/mon/mon.go#L388) should let the [kubernetes scheduler](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-scheduler/) decide more where to put the mon pod.

The current behaviour makes setting resource requests/limits of ceph-components useless
regarding "finding the best k8s node with sufficient resources"
as they have no impact on on which node to schedule the mon pod.
The [scheduleMonitor func](https://github.com/rook/rook/blob/50ef726ad34da4f2683d5f4304eb4e7910bdb938/pkg/operator/ceph/cluster/mon/mon.go#L388) which is getting a node for mon is not even considering
if the node has enough resources to allocate the mon pod. Later on the resulting mon deployment spec gets the wrong node returned from that func containing the node selector for a  node that cannot schedule the mon pod **while other nodes could**. 
```
      nodeSelector:
        kubernetes.io/hostname: nodeC
```
The [kubernetes scheduler](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-scheduler/) should be more involved in the node selecting decission. 

**How to reproduce it (minimal and precise):**
A kubernetes cluster of 3 nodes where some are able to fulfill mon pods resource requirements and nodes that are not.

```
nodeA - 2 CPU & 8 GB RAM
nodeB - 2 CPU & 4 GB RAM
nodeC - 2 CPU & 4 GB RAM
```
This should schedule a mon pod with 6 GB RAM on nodeA, which the operator is not doing and picking a "random" node.

A mon that requests the amount of resources that can be scheduled on nodeA but not on nodeB and nodeC.
```yaml
  resources:
    mon:
      requests:
        cpu: "500m"
        memory: "6Gib"
      limits:
        cpu: "500m"
        memory: "6Gib"
```

**Comment by travisn** [comment]
#3475 would take this issue into account

**Comment by robinbraemer** [comment]
Also when changing mon count of the cluster the operator is not removing those mons that are ina pending state (mons that can't be scheduled).

**Comment by noahdesu** [comment]
@robinbraemer can you please create a ticket for your new observation?

**Comment by noahdesu** [comment]
@travisn did this behavior ever work in which the mon scheduler was looking for nodes that satisfied the resource constraint in the crd? I do not recall encountering this when reworking the scheduler, but if so, perhaps I missed something and need add this back in.

**Comment by travisn** [comment]
@noahdesu No, I don't believe the resource constraints have been taken into account when rook is scheduling the mons, so I would expect this issue is not new. But assuming we can rely on k8s for mon scheduling, we would get this for free of course.

**Comment by travisn** [comment]
This was taken into account with #3668 and was included in v1.1.
